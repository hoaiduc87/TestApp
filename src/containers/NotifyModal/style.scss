@import 'assets/styles/mixins.scss';

.modal {
  &--error {
    text-align: center;
    background-color: $white;

    &__header {
      padding: 16px;
      font-size: 18px;
      background-color: $red;
      color: white;
    }

    &__body {
      padding: 20px;
    }
  }
}

@include mobile {
  .modal {
    &--error {
      &__header {
        padding: 8px;
        font-size: 14px;
      }

      &__body {
        padding: 12px;
        font-size: 12px;
      }
    }
  }
}
