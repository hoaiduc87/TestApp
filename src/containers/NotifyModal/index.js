import React, { memo } from 'react';
import Modal from 'components/Modal';
import { connect } from 'react-redux';
import { compose } from 'redux';
import { createStructuredSelector } from 'reselect';

import { setNotifyModal } from 'containers/App/actions';
import { selectNotifyModal } from 'containers/App/selectors';

import './style.scss';

function NotifyModal({
  notifyModal,
  setNotifyModalRequest,
}) {
  const handleClose = () => {
    setNotifyModalRequest(null);
  };

  const type = notifyModal && notifyModal.type;

  return (
    <Modal show={!!notifyModal} className={`modal--small modal--${type}`} onClose={handleClose}>
      {!!notifyModal &&
        <>
          <div className={`modal--${type}__header`}>{notifyModal.title}</div>
          <div className={`modal--${type}__body`}>{notifyModal.message}</div>
        </>
      }
    </Modal>
  )
}

const mapStateToProps = createStructuredSelector({
  notifyModal: selectNotifyModal(),
});

const mapDispatchToProps = (dispatch) => ({
  setNotifyModalRequest: data => dispatch(setNotifyModal(data)),
});

const withConnect = connect(mapStateToProps, mapDispatchToProps);

export default compose(
  withConnect,
  memo,
)(NotifyModal);
