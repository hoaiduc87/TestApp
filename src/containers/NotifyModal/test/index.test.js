import React from 'react';
import { render } from '@testing-library/react';
import { mount } from 'enzyme';
import configureMockStore from 'redux-mock-store';
import { Provider } from 'react-redux';

import NotifyModal from '..';

const mockStore = configureMockStore();
const initialState = {
  app: {
    notifyModal: {
      type: 'error',
      title: 'Error',
      message: 'Error occur'
    },
  }
};
const store = mockStore(initialState);

describe('<NotifyModal/>', () => {
  it('Should run and match snapshot', () => {
    const Wrapper = ({children}) => <Provider store={store}>{children}</Provider>;
    const {
      container: { firstChild }
    } = render(<NotifyModal />, { wrapper: Wrapper });
    expect(firstChild).toMatchSnapshot();
  });

  it('Should execute setNotifyModalRequest once click close button', () => {
    store.dispatch = jest.fn();
    const comp = mount(
      <Provider store={store}>
        <NotifyModal />
      </Provider>
    );
    comp.find('.modal__dialog__close').first().simulate('click');
    expect(store.dispatch).toHaveBeenCalled();
  });
})