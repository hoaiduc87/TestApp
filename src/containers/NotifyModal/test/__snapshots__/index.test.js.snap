// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NotifyModal/> Should run and match snapshot 1`] = `
<div
  class="modal modal--visible"
>
  <div
    class="modal__dialog modal--small modal--error"
  >
    <span
      class="modal__dialog__close"
    >
      <img
        alt="close"
        src="close_white.svg"
      />
    </span>
    <div
      class="modal--error__header"
    >
      Error
    </div>
    <div
      class="modal--error__body"
    >
      Error occur
    </div>
  </div>
  <div
    class="modal__backdrop"
  />
</div>
`;
